### PR TITLE
[ISSUE #3141]  add producer acl authentication

### LINKED ISSUE #3141: [Enhancement] Added token-based ACL authentication, which uses the publickey in the key pair for verification
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Supports token-based authentication to verify the token string in the request

### Describe the solution you'd like

Supports token-based authentication to verify the token string in the request

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by slowpao**
Fixes #3141 .

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by mytang0** [comment]
Please fix the license check error
image

File: eventmesh-registry-plugin/eventmesh-registry-api/src/main/java/org/apache/eventmesh/api/registry/bo/EventMeshServicePubTopicInfo.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "EventMeshServicePubTopicInfo{"
+            + "service='" + service + '\''
+            + ", topics=" + topics
+            + '}';
+    }
+}
```
Comment (mytang0) [inline eventmesh-registry-plugin/eventmesh-registry-api/src/main/java/org/apache/eventmesh/api/registry/bo/EventMeshServicePubTopicInfo.java:59]: It is recommended to add code comments

File: eventmesh-common/src/test/java/org/apache/eventmesh/common/file/WatchFileManagerTest.java [hunk 2]
```diff
@@ -33,29 +36,36 @@
...
+        try (BufferedReader reader = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
+            properties.load(reader);
+        } catch (IOException e) {
+            Assert.fail("Test failed to load from file");
+        }
         properties.setProperty("eventMesh.server.newAdd", "newAdd");
-        FileWriter fw = new FileWriter(file);
-        properties.store(fw, "newAdd");
-
+        try (BufferedWriter writer = Files.newBufferedWriter(path, StandardCharsets.UTF_8)) {
```
[the remark this answers is not in the document]
Comment (slowpao) [reply eventmesh-common/src/test/java/org/apache/eventmesh/common/file/WatchFileManagerTest.java:64]: OK, I will fix it next time I submit

File: eventmesh-registry-plugin/eventmesh-registry-api/src/main/java/org/apache/eventmesh/api/registry/bo/EventMeshServicePubTopicInfo.java (hunk 1 above)
Comment (slowpao) [reply eventmesh-registry-plugin/eventmesh-registry-api/src/main/java/org/apache/eventmesh/api/registry/bo/EventMeshServicePubTopicInfo.java:59]: OK, I will fix it next time I submit

**Pull request merged**